src/plugins/discover/public/application/main/hooks/use_discover_state.ts

```diff
@@ -226,6 +249,41 @@ export function useDiscoverState({
     }
   }, [initialFetchStatus, refetch$, indexPattern, savedSearch.id]);
 
+  const fetchResults = useCallback(() => {
+    if (documentState.result?.length) {
+      const firstRow = documentState.result[0];
+      const columns = Object.keys(firstRow.raw).slice(0, MAX_NUM_OF_COLUMNS);
+      if (!isEqual(columns, documentStateCols) && !isEqual(state.query, sqlQuery)) {
+        return columns;
+      }
+      return [];
+    }
+    return [];
+  }, [documentState.result, documentStateCols, sqlQuery, state.query]);
+
+  useEffect(() => {
+    async function fetchDataview() {
+      if (state.query && isOfAggregateQueryType(state.query)) {
+        const indexPatternFROMQuery = getIndexPatternFromSQLQuery(state.query.sql);
+        const idsTitles = await indexPatterns.getIdsWithTitle();
+        const dataViewObj = idsTitles.find(({ title }) => title === indexPatternFROMQuery);
```

Comment:
Nit, because if tend to read FROMQuery ... something like a SQLQuery 😄 
```suggestion
        const indexPatternFromQuery = getIndexPatternFromSQLQuery(state.query.sql);
        const idsTitles = await indexPatterns.getIdsWithTitle();
        const dataViewObj = idsTitles.find(({ title }) => title === indexPatternFromQuery);
```